validate_sample.py

```diff
@@ -0,0 +1,339 @@
+        validate_htc_willingness(line_number, sample_row)
+        validate_htc_digital(line_number, sample_row)
+        validate_fieldcordinator_id(line_number, sample_row)
+        validate_fieldofficer_id(line_number, sample_row)
+        validate_treatment_code(line_number, sample_row)
+        validate_ce_expected_capacity(line_number, sample_row)
+
+
+def validate_arid(line_number, sample_row):
+    global error_count
+    column = 'ARID'
+    maximum_length = 21
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+    if sample_row[column] in ARID:
+        print(f'Line {line_number}: {column}: {sample_row[column]} is duplicated in sample file.')
+        error_count += 1
+    else:
+        ARID.add(sample_row[column])
+
+
+def validate_estab_arid(line_number, sample_row):
+    column = 'ESTAB_ARID'
+    maximum_length = 21
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+
+
+def validate_uprn(line_number, sample_row):
+    global error_count
+    column = 'UPRN'
+    maximum_length = 12
+    if _check_mandatory_value_exists(line_number, column, mandatory=False, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+        uprn = sample_row[column]
+        if not uprn.isnumeric() and uprn != '':
+            print(f'Line {line_number}: {column}: {sample_row[column]} is not a valid integer.')
+            error_count += 1
+
+
+def validate_address_type(line_number, sample_row):
+    global error_count
+    column = 'ADDRESS_TYPE'
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        address_type = sample_row[column]
+        if address_type not in {"HH", "CE", "SPG"}:
+            print(f'Line {line_number}: {column}: {sample_row[column]} is not valid.')
+            error_count += 1
+
+
+def validate_estab_type(line_number, sample_row):
+    column = 'ESTAB_TYPE'
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _is_valid_estab_type(line_number, sample_row[column])
+
+
+def validate_address_level(line_number, sample_row):
+    global error_count
+    column = 'ADDRESS_LEVEL'
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        if sample_row[column] not in {"E", "U"}:
+            print(f'Line {line_number}: {column}: {sample_row[column]} is not valid.')
+            error_count += 1
+
+
+def validate_abp_code(line_number, sample_row):
+    column = 'ABP_CODE'
+    maximum_length = 6
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
```

Comment:
This exact pattern is repeated several times, it could be changed to a single function which is passed the column string.

Reply:
Agree it could be changed (a lot of this stuff could still be refactored) but this should be probably weighed up against how much more time we're willing to spend on a script that works and is only likely to be used once ever.

Reply:
Fair point and I won't block the PR on this, but even for scripts like this I generally consider it worth a bit of effort to avoid code repetition and anti patterns as they only cause more pain if you do have to come back to it in the future!